staging/src/k8s.io/apiserver/pkg/audit/request_test.go

```diff
@@ -54,3 +60,166 @@ func TestMaybeTruncateUserAgent(t *testing.T) {
+							ManagedFields: []metav1.ManagedFieldsEntry{
+								{Manager: "c", Operation: metav1.ManagedFieldsOperationUpdate, Time: &metav1.Time{Time: time.Now()}},
+								{Manager: "d", Operation: metav1.ManagedFieldsOperationUpdate, Time: &metav1.Time{Time: time.Now()}},
+							},
+						},
+					},
+				},
+			},
+			expected: &corev1.PodList{
+				Items: []corev1.Pod{
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "foo",
+							Namespace: "ns1",
+						},
+					},
+					{
+						ObjectMeta: metav1.ObjectMeta{
+							Name:      "bar",
+							Namespace: "ns2",
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "object specified is a Table and objects in its rows have managed fields",
+			object: &metav1.Table{
+				Rows: []metav1.TableRow{
+					{
+						Object: runtime.RawExtension{
+							Object: &corev1.Pod{
+								ObjectMeta: metav1.ObjectMeta{
+									Name:      "foo",
+									Namespace: "ns1",
+									ManagedFields: []metav1.ManagedFieldsEntry{
+										{Manager: "a", Operation: metav1.ManagedFieldsOperationUpdate, Time: &metav1.Time{Time: time.Now()}},
+										{Manager: "b", Operation: metav1.ManagedFieldsOperationUpdate, Time: &metav1.Time{Time: time.Now()}},
+									},
+								},
+							},
+						},
+					},
+					{
+						Object: runtime.RawExtension{
+							Object: &corev1.Pod{
+								ObjectMeta: metav1.ObjectMeta{
+									Name:      "bar",
+									Namespace: "ns2",
+									ManagedFields: []metav1.ManagedFieldsEntry{
+										{Manager: "c", Operation: metav1.ManagedFieldsOperationUpdate, Time: &metav1.Time{Time: time.Now()}},
+										{Manager: "d", Operation: metav1.ManagedFieldsOperationUpdate, Time: &metav1.Time{Time: time.Now()}},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			expected: &metav1.Table{
+				Rows: []metav1.TableRow{
+					{
+						Object: runtime.RawExtension{
+							Object: &corev1.Pod{
+								ObjectMeta: metav1.ObjectMeta{
+									Name:      "foo",
+									Namespace: "ns1",
+								},
+							},
+						},
+					},
+					{
+						Object: runtime.RawExtension{
+							Object: &corev1.Pod{
+								ObjectMeta: metav1.ObjectMeta{
+									Name:      "bar",
+									Namespace: "ns2",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			objectGot, ok, err := copyWithoutManagedFields(test.object)
+
+			if test.expected == nil {
+				if err != nil {
+					t.Errorf("expected error to be nil, but got %v", err)
+				}
+				if ok {
+					t.Error("expected ok to be false, but got true")
+				}
+				if objectGot != nil {
+					t.Errorf("expected the object returned to be nil, but got %v", objectGot)
+				}
+				return
+			}
+
+			if err != nil {
+				t.Errorf("expected error to be nil, but got %v", err)
+			}
+			if !ok {
+				t.Error("expected ok to be true, but got false")
+			}
+			if !reflect.DeepEqual(test.expected, objectGot) {
```

Comment:
We use `cmp.Equal` nowadays, I think.

This tests the "copy" part but not the "deep" part-- if that's important it should be verified. (And if it's not important it should not be done!)

Reply:
+1 to `cmp.Equal` 
I added a step that verifies that deep copy is done.